src/scripts/settings.js

```diff
@@ -664,12 +638,14 @@ document.addEventListener('DOMContentLoaded', function(e) {
         Settings.$permissionFilterClear.style.display = 'none';
       }
       Settings.lastFilter = val;
-      for (key in Settings.permissionItems) {
-        if (Settings.permissionItems.hasOwnProperty(key)) {
-          if (Settings.permissionItems[key].id.indexOf(val) !== -1) {
-            Settings.permissionItems[key].classList.add('filter');
-          } else if (!!Settings.permissionItems[key].classList) {
-            Settings.permissionItems[key].classList.remove('filter');
+
+      var permissionItems = document.querySelectorAll('#permissions-list li');
```

Comment:
We can modernize some of the code a bit

```javascript
const permissionItems = [...document.querySelectorAll('#permissions-list li')];
permissonItems.forEach((item) => {
    //
});
```

begone `hasOwnProperty`

Reply:
I must say I actually prefer the original, it's pretty unclear what the spread does.

Reply:
Actually the `forEach` function is actually present on `NodeList`, so can use it directly.

The `...` converts the `NodeList<Element>` to an array of `Element`.
Can also be written as
```javascript
const permissionItems = Array.from(document.querySelectorAll('#permissions-list li');
```